tests/v2/validation/rbac/clusterandprojectroles/project_role_test.go

```diff
@@ -0,0 +1,121 @@
+//go:build (validation || infra.any || cluster.any || extended) && !sanity && !stress
+
+package roles
+
+import (
+	"testing"
+
+	"github.com/rancher/shepherd/clients/rancher"
+	management "github.com/rancher/shepherd/clients/rancher/generated/management/v3"
+	"github.com/rancher/shepherd/extensions/clusters"
+	"github.com/rancher/shepherd/extensions/projects"
+	rbac "github.com/rancher/shepherd/extensions/rbac"
+	"github.com/rancher/shepherd/extensions/users"
+	"github.com/rancher/shepherd/pkg/session"
+	log "github.com/sirupsen/logrus"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+)
+
+type ProjectRolesTestSuite struct {
+	suite.Suite
+	client  *rancher.Client
+	session *session.Session
+	cluster *management.Cluster
+}
+
+func (pr *ProjectRolesTestSuite) TearDownSuite() {
+	pr.session.Cleanup()
+}
+
+func (pr *ProjectRolesTestSuite) SetupSuite() {
+	testSession := session.NewSession()
+	pr.session = testSession
+
+	client, err := rancher.NewClient("", testSession)
+	require.NoError(pr.T(), err)
+
+	pr.client = client
+
+	log.Info("Getting cluster name from the config file and append cluster details in pr")
+	clusterName := client.RancherConfig.ClusterName
+	require.NotEmptyf(pr.T(), clusterName, "Cluster name to install should be set")
+	clusterID, err := clusters.GetClusterIDByName(pr.client, clusterName)
+	require.NoError(pr.T(), err, "Error getting cluster ID")
+	pr.cluster, err = pr.client.Management.Cluster.ByID(clusterID)
+	require.NoError(pr.T(), err)
+}
+
+func (pr *ProjectRolesTestSuite) testSetupUserAndProject() (*rancher.Client, *management.Project) {
+	pr.T().Log("Set up User with cluster Role for additional rbac test cases " + rbac.ClusterOwner)
+	newUser, standardUserClient, err := rbac.SetupUser(pr.client, rbac.StandardUser.String())
+	require.NoError(pr.T(), err)
+
+	createProjectAsAdmin, err := pr.client.Management.Project.Create(projects.NewProjectConfig(pr.cluster.ID))
```

Comment:
Now that we have public api available for creating a project, should we consider modifying this to use that instead of norman? Since this PR is to refactor the structure, are you planning to create a new PR to change this to use public API instead of norman?

Reply:
Since this is a refactoring or restructuring of the tests, will not be fixing the changes from norman to public API in this PR. Created this task https://github.com/rancher/qa-tasks/issues/1305